src/components/Chat/Sidebar/Sidebar.jsx

```diff
@@ -1,10 +1,22 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { SidebarContainer, SidebarSection, RouteLink, SidebarTitle, UserList, SidebarHeader } from "./SidebarElements";
 import Contact from "../Contact";
 import SearchInputBox from "../../Common/SearchInputBox";
 import { FaAngleDown, FaAngleUp } from "react-icons/fa";
+import { useDispatch, useSelector } from "react-redux";
+import { getConnections } from "../../../features/connections/connectionSlice";
 
 const Sidebar = ({ hideSidebar, onlinePeople, offlinePeople, selectedUserId, setSelectedUserId }) => {
+    const dispatch = useDispatch();
+    const { user } = useSelector((state) => state.auth);
+    const connections = useSelector((state) => state.connectionData);
+    const allConnections = connections?.connections?.connections?.map((connection) => connection.user);
```

Comment:
You forgot to check if `isAccepted` is true so when user only sends connection request, the "connection user" appear in the chat before the other side accepting or rejecting the request.